instant-distance/src/lib.rs

```diff
@@ -394,6 +400,55 @@ where
     pub fn get(&self, i: usize, search: &Search) -> Option<Item<'_, P>> {
         Some(Item::new(search.nearest.get(i).copied()?, self))
     }
+
+    pub fn insert(
+        &mut self,
+        point: P,
+        ef_construction: usize,
+        heuristic: Option<Heuristic>,
+    ) -> PointId {
+        let new_pid = self.points.len();
+        let new_point_id = PointId(new_pid as u32);
+
+        self.points.push(point);
+        self.zero.push(ZeroNode::default());
+
+        let zeros = self
+            .zero
+            .iter()
+            .map(|z| RwLock::new(z.clone()))
+            .collect::<Vec<_>>();
```

Comment:
This seems like a pretty expensive operation, to the point that calling `insert()` one-by-one is unlikely to be very effective. Could we yield some construction type here that can be reused to insert multiple points?